### PR TITLE
fix: Reuse CometBroadcastExchangeExec with Spark ReuseExchangeAndSubquery rule

### LINKED ISSUE #439: `CometBroadcastExchangeExec` cannot be reused by Spark `ReuseExchangeAndSubquery` rule
### Describe the bug

Spark `ReuseExchangeAndSubquery` rule detects duplicate exchanges in a query plan and reuses them if possible. When working on #437, I found that some Spark tests are failed because `CometBroadcastExchangeExec` cannot be reused by the rule.

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #439.

### CODE REVIEW

**Comment by viirya** [comment]
cc @andygrove @sunchao

File: spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala [hunk 1]
```diff
@@ -62,6 +63,39 @@ class CometExecSuite extends CometTestBase {
     }
   }
 
+  test("ReusedExchangeExec should work on CometBroadcastExchangeExec") {
```
Comment (viirya) [inline spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala:66]: Hmm, this test only fails on Spark 3.3. Let me take a look.

File: spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala [hunk 2]
```diff
@@ -576,11 +576,13 @@ class CometSparkSessionExtensions
...
         // exchange to Comet broadcast exchange if its downstream is a Comet native plan or if the
         // broadcast exchange is forced to be enabled by Comet config.
+        // Note that `CometBroadcastExchangeExec` is only supported for Spark 3.4+.
         case plan if plan.children.exists(_.isInstanceOf[BroadcastExchangeExec]) =>
           val newChildren = plan.children.map {
             case b: BroadcastExchangeExec
                 if isCometNative(b.child) &&
-                  isCometOperatorEnabled(conf, "broadcastExchangeExec") =>
+                  isCometOperatorEnabled(conf, "broadcastExchangeExec") &&
+                  isSpark34Plus => // Spark 3.4+ only
```
Comment (kazuyukitanimura) [inline spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala:585]: If `BroadcastExchangeExec` is only for 3.4+, is `isSpark34Plus` extra?

**Review by kazuyukitanimura — APPROVED** [review]
LGTM

File: spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala (hunk 2 above)
Comment (viirya) [reply spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala:585]: `CometBroadcastExchangeExec` is only for 3.4+ not `BroadcastExchangeExec`.

**Review by sunchao — APPROVED** [review]

**Comment by viirya** [comment]
Merged. Thanks @kazuyukitanimura @sunchao

**Pull request merged**